### PR TITLE
Readme dependencies

### LINKED ISSUE #274: Mention autoconf dependency (aclocal) in the build-from-source documentation
Attempting to build on a fresh Debian system, `./bootstrap.sh` failed citing not recognizing `aclocal` as a proper command. I discovered this is included in the Debian `autoconf` package, which is not referred to in any ZNC build guides. Perhaps this should be added?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98417-mention-autoconf-dependency-aclocal-in-the-build-from-source-documentation?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #747: README not listing all dependencies, swap for a link to wiki?
https://github.com/znc/znc#minimal-requirements

Misses `pkg-config` and `g++`.

@KindOne- mentioned in #znc that the installation stuff could be removed and exchanged for a link to http://wiki.znc.in/Installation .

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/5927543-readme-not-listing-all-dependencies-swap-for-a-link-to-wiki?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #834: Add note about libicu-dev to build instructions
<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8177710-add-note-about-libicu-dev-to-build-instructions?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Kriechi**
[ci skip]
closes #747
closes #834
closes #274

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,53 +17,63 @@
 
 Core:
 
- - GNU make (try gmake if make fails)
- - GCC 4 or later
+* GNU make
+* autoconf and automake
+* pkg-config
```
Comment (DarthGandalf) [inline README.md]: autoconf/automake is needed only if you're building from git, not from tarball

File: README.md [hunk 2]
```diff
@@ -17,53 +17,63 @@
...
+* GCC 4 or clang
```
Comment (DarthGandalf) [inline README.md]: GCC 4.7 or clang 3.2

File: README.md [hunk 3]
```diff
@@ -17,53 +17,63 @@
...
 
 ## Installing ZNC
 
-If you are building from git, you will need to run `./autogen.sh` first to produce the `configure` script.
-Note that this requires `automake` and `gettext` to be installed.
+Installation from source code is performed via `automake` toolchain:
 
-Installation is done with the `./configure ; make ; make install` commands.
+```shell
+./autogen.sh
```
Comment (DarthGandalf) [inline README.md]: releases and nightlies already have working configure

File: README.md [hunk 4]
```diff
@@ -17,53 +17,63 @@
...
 modperl:
-
- - This needs perl and its bundled libperl
+* perl and its bundled libperl
 
 modpython:
-
- - This needs perl(!) and python's bundled libpython
+* python and its bundled libpython
+* modperl is required
```
Comment (DarthGandalf) [inline README.md]: nope, modperl not required. perl is.

File: README.md [hunk 5]
```diff
@@ -17,53 +17,63 @@
...
 
 ## Optional Requirements
 
-SSL support:
-
- - openssl 0.9.7d or later (try installing openssl-dev, openssl-devel or
-   libssl-dev)
+SSL/TLS support:
+* openssl 0.9.7d or later
+    * try installing openssl*dev, openssl*devel or libssl*dev
```
Comment (DarthGandalf) [inline README.md]: what is `*`?

File: README.md [hunk 6]
```diff
@@ -17,53 +17,63 @@
...
 
 ## Setting up znc.conf
 
-For setting up a configuration file in `~/.znc` you can simply do
-	znc --makeconf
-or
-	./znc --makeconf
-for in-place execution.
+For setting up a configuration file in `~/.znc`:
+```shell
```
Comment (DarthGandalf) [inline README.md]: The benefit of using markdown was that it looks beautiful in plain text too, not only after rendering :(
All the "build status" images and such code blocks breaks it...

File: README.md [hunk 7]
```diff
@@ -17,53 +17,63 @@
...
+SSL/TLS support:
+* openssl 0.9.7d or later
+    * try installing openssl*dev, openssl*devel or libssl*dev
+    * Mac OS X: OpenSSL from Homebrew is prefered over system
+        * use `--USE_SYSTEM_OPENSSL` to force `configure` to use the (deprecated) Mac OS X version
 
 modperl:
-
- - This needs perl and its bundled libperl
+* perl and its bundled libperl
```
Comment (DarthGandalf) [inline README.md]: Also SWIG for both modperl and modpython, but only for building from git.

Comment (Kriechi) [inline README.md]: ok. I added it.

**Comment by Kriechi** [comment]
@DarthGandalf I made all changes you commented about.

File: README.md [hunk 8]
```diff
@@ -17,53 +17,59 @@
...
 
 ## Optional Requirements
 
-SSL support:
-
- - openssl 0.9.7d or later (try installing openssl-dev, openssl-devel or
-   libssl-dev)
+SSL/TLS support:
+* openssl 0.9.7d or later
+    * try installing openssl*dev, openssl-devel or libssl-dev
```
Comment (DarthGandalf) [inline README.md]: What is this?

Comment (Kriechi) [inline README.md]: sorry - my editor went rogue...
should be fixed now.

File: README.md [hunk 9]
```diff
@@ -17,53 +17,59 @@
...
 
-SSL support:
-
- - openssl 0.9.7d or later (try installing openssl-dev, openssl-devel or
-   libssl-dev)
+SSL/TLS support:
+* openssl 0.9.7d or later
+    * try installing openssl-dev, openssl-devel or libssl-dev
+    * Mac OS X: OpenSSL from Homebrew is prefered over system
+        * use `--USE_SYSTEM_OPENSSL` to force `configure` to use the (deprecated) Mac OS X version
```
Comment (DarthGandalf) [inline README.md]: I thought it was env variable, not a `--` option?

Comment (Kriechi) [inline README.md]: good catch - yes.
I changed it.

File: README.md [hunk 10]
```diff
@@ -17,53 +17,59 @@
...
 
 You can use
-	./configure --help
+	`./configure --help`
 if you want to get a list of options, though the defaults should be suiting
 most needs. After you compiled it with make (or gmake if make doesn't work) you
 can install it with
-	make install
+	`make install`
 though you don't need to as ZNC supports in-place execution.
```
Comment (DarthGandalf) [inline README.md]: Oh, this was loong ago. It doesn't support that...

Comment (Kriechi) [inline README.md]: no more in-place execution? I'll remove that part then.

**Pull request merged**